### PR TITLE
Fix check for network restart in yast2_lan

### PR DESCRIPTION
**Pull request by chcao**
Add code to handle the cancel button for network testing. And modify the yast2_lan_restart* test modules by using the cancel button instead of next button.

- Related ticket: https://progress.opensuse.org/issues/120639
- Needles: n/a
- Verification run: 
https://openqa.suse.de/tests/10149238#

### CODE REVIEW

File: lib/YaST/NetworkSettings/v4_3/NetworkSettingsController.pm [hunk 1]
```diff
@@ -85,6 +86,21 @@ sub save_changes {
...
 
+sub cancel_changes {
+    my ($self) = @_;
+    $self->get_action_buttons()->press_cancel();
+    $self->get_cancel_changes_warning()->press_yes();
+}
+
+sub get_cancel_changes_warning {
+    my ($self) = @_;
+    if ($self->{CancelChangesWarning}->text() !~
```
[the remark this answers is not in the document]
Comment (jknphy) [reply lib/YaST/NetworkSettings/v4_3/NetworkSettingsController.pm:97]: Jonathan is right but the problem here is that, well first this libyui-rest-api code was done before installation...so it is really early stage, no so organized, second that the page in this case is the popup which we reuse, so not all yes/no popup has the same text, we could inherit and problem solved, but that would create too many children popups...
I think the best solution is avoid checking text at all that can be read in the screen (as we usually do, because we check by command the important things later), so if the popup is shown, we accept it and that's it. That should work I guess.

To do it better in the future we should have some call from the Test layer, so it would be:
```
- cancel()
- accept_all_changes_will_be_lost()
```
`accept_all_changes_will_be_lost()` could have some this logic with the regex.

In other words, `$self->get_cancel_changes_warning()` should just check the popup without more complex actions as it is a getter. `accept_all_changes_will_be_lost()` can be called and use inside something  like:
`$self->get_cancel_changes_warning()->text() ...` and do the regex.

The thing is that Controller Layer and Page layer could have been in ONE layer, this is some over-complication we did, so check of the text we could do it in the Controller but leaving the method simple enough.

Let me know if it is not clear enough :) it is one of those corner cases, you could try some modification see how it looks, but if we find more complication we could merge this as we might revisit this old libyui-rest-code in the future.

[the remark this answers is not in the document]
Comment (chcao) [reply lib/YaST/NetworkSettings/v4_3/NetworkSettingsController.pm]: Thanks for review. I updated the pr, please help to take a look at it, thanks.
